src/Tools/dotnet-monitor/Egress/EgressProviderInternal.cs

```diff
@@ -20,7 +21,7 @@ namespace Microsoft.Diagnostics.Tools.Monitor.Egress
     /// </summary>
     internal class EgressProviderInternal<TOptions> :
         IEgressProviderInternal<TOptions>
-        where TOptions : class
+        where TOptions : new()
```

Comment:
Why the constraint change?

Reply:
This is required because the manual enumeration of `TOptions` requires the class to make a new instance of `TOptions` to bind to. Normally this was done via reflection by the IConfiguraitonOptions internal goo, but adding this constraint makes it easy to do manually.

Reply:
I don't see why this is change is necessary here.

Reply:
Ahh, this isn't required now that `IEgressProviderConfigurationProvider` doesn't do the `TOptions` creation anymore.